src/utils/viewer/viewer.cpp

```diff
@@ -515,9 +574,19 @@ void Viewer::keyPressEvent(KeyEvent& event) {
   }
   renderCamera_->node().setTransformation(
       rgbSensorNode_->absoluteTransformation());
+
+  // XXX This will cause segfault
+  updateText();
```

Comment:
backtrace?

Reply:
@mosra : The bt is as follows:

I noticed that it says "TextureTools::atlas(): requested atlas size Vector(512, 512) is too small to fit 70 Vector(90, 108) textures. Generated atlas will be empty.". Would the segfault be caused by this?

```
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1111 12:19:44.038406 16127424 viewer.cpp:210] Loading navmesh from /Users/yili/models/data/scene_datasets/habitat-test-scenes/skokloster-castle.navmesh
TextureTools::atlas(): requested atlas size Vector(512, 512) is too small to fit 70 Vector(90, 108) textures. Generated atlas will be empty.
I1111 12:19:47.641803 16127424 viewer.cpp:221] position=0-5.14033 0.0099186 0024.1693 rotation=0000000000 -0.0980171 0000000000 000.995185
I1111 12:19:47.641839 16127424 viewer.cpp:223] Distance to closest obstacle: 0.19586
viewer was compiled with optimization - stepping may behave oddly; variables may not be available.
Process 29100 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x30)
    frame #0: 0x00000001000557af viewer`Magnum::Text::AbstractFont::lineHeight() const [inlined] Magnum::Text::AbstractFont::isOpened(this=0x0000000101033e40) const at AbstractFont.h:281:40 [opt]
   278          #endif
   279
   280          /** @brief Whether any file is opened */
-> 281          bool isOpened() const { return doIsOpened(); }
   282
   283          /**
   284           * @brief Open font from raw data
Target 0: (viewer) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x30)
  * frame #0: 0x00000001000557af viewer`Magnum::Text::AbstractFont::lineHeight() const [inlined] Magnum::Text::AbstractFont::isOpened(this=0x0000000101033e40) const at AbstractFont.h:281:40 [opt]
    frame #1: 0x00000001000557ac viewer`Magnum::Text::AbstractFont::lineHeight(this=0x0000000101033e40) const at AbstractFont.cpp:229 [opt]
    frame #2: 0x0000000100053b42 viewer`Magnum::Text::(anonymous namespace)::renderVerticesInternal(font=0x0000000101033e40, cache=0x00007ffeefbff438, size=0, text="FPS: 6e+01", alignment=TopRight) at Renderer.cpp:79:53 [opt]
    frame #3: 0x0000000100054963 viewer`Magnum::Text::AbstractRenderer::render(this=0x0000000101032d10, text=<unavailable>) at Renderer.cpp:364:40 [opt]
    frame #4: 0x000000010000cf63 viewer`(anonymous namespace)::Viewer::updateText(this=<unavailable>) at viewer.cpp:587:21 [opt]
    frame #5: 0x0000000100008759 viewer`(anonymous namespace)::Viewer::keyPressEvent(this=0x00007ffeefbff170, event=<unavailable>) at viewer.cpp:579:3 [opt]
    frame #6: 0x00000001004a4482 viewer`Magnum::Platform::GlfwApplication::setupCallbacks()::$_4::__invoke(GLFWwindow*, int, int, int, int) [inlined] Magnum::Platform::GlfwApplication::setupCallbacks(window=<unavailable>, key=262, (null)=0, action=1, mods=0)::$_4::operator()(GLFWwindow*, int, int, int, int) const at GlfwApplication.cpp:507:17 [opt]
    frame #7: 0x00000001004a4455 viewer`Magnum::Platform::GlfwApplication::setupCallbacks(window=<unavailable>, key=262, (null)=<unavailable>, action=1, mods=0)::$_4::__invoke(GLFWwindow*, int, int, int, int) at GlfwApplication.cpp:501 [opt]
    frame #8: 0x00000001004af037 viewer`-[GLFWContentView keyDown:](self=0x000000010162ae70, _cmd=<unavailable>, event=0x000000010143c7f0) at cocoa_window.m:663:5 [opt]
    frame #9: 0x00007fff36839367 AppKit`-[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 6840
    frame #10: 0x00007fff36837667 AppKit`-[NSWindow(NSEventRouting) sendEvent:] + 478
    frame #11: 0x00007fff366d7889 AppKit`-[NSApplication(NSEvent) sendEvent:] + 2953
    frame #12: 0x00000001004afa62 viewer`-[GLFWApplication sendEvent:](self=0x0000000101119c80, _cmd=<unavailable>, event=0x000000010143c7f0) at cocoa_window.m:904:9 [opt]
    frame #13: 0x00000001004b18e7 viewer`_glfwPlatformPollEvents at cocoa_window.m:1590:9 [opt]
    frame #14: 0x00000001004a3ca5 viewer`Magnum::Platform::GlfwApplication::exec(this=0x00007ffeefbff170) at GlfwApplication.cpp:601:9 [opt]
    frame #15: 0x0000000100005a36 viewer`main(argc=2, argv=<unavailable>) at viewer.cpp:592:1 [opt]
    frame #16: 0x00007fff650383d5 libdyld.dylib`start + 1
```

Reply:
Looks to be because the `font` passed to the `Text::Renderer2D` constructor gets deleted at the end of Viewer constructor. You need to save it as a class member, together with the `fontManager`

Reply:
And re the warning message -- try to have the font size only e.g. 16, not 110. 110 was needed for the text example because there it's *huge*, here you don't need that much. Play with the value until it looks sharp on the screen -- best results are when the font size is exactly 2x larger than what appears on the screen (so, 2x supersampled).